app/models/registration.rb

```diff
@@ -46,16 +46,19 @@ class Registration < ApplicationRecord
 
   serialize :roles, coder: YAML
 
+  # Set a `registered_at` timestamp for newly created records,
+  #   but only if there is no value already specified from the outside
+  after_initialize :mark_registered_at, if: :new_record?, unless: :registered_at?
+  before_create -> { self.registrant_id ||= competition.registrations.count + 1 }
+  validates :registrant_id, uniqueness: { scope: :competition_id }
```

Comment:
Can this validation actually screw us over as long as the values are not backfilled yet?

Reply:
I mean, during the very initial deployment, it will see a bunch of `NULL`s and probably freak out

Reply:
All I can say is that it works locally, but I'm happy to remove it/put it behind an environment flag until we're done with backfilling